llvm/lib/Transforms/Yk/LivenessAnalysis.cpp

```diff
@@ -121,11 +121,11 @@ LivenessAnalysis::LivenessAnalysis(Function *Func) {
 
   // A function implicitly defines its arguments.
   //
-  // To propagate the arguments properly we pretend that the first instruction
-  // in the entry block defines the arguments.
+  // To propagate the arguments properly we say that the function's arguments
```

Comment:
I can't parse this sentence.

Reply:
This is essentially saying, that in order for the function arguments to be live, the liveness algo needs to pretend that the function arguments are incoming values into the first block. It's just the way the algo works.

Reply:
That's fine: I just can't parse the English ;)

Reply:
Which part Laurie?

Reply:
The sentence.

Reply:
"incoming to" -> "flow into"?

Reply:
Edd pointed out that my comments was, ahem, not entirely helpful. Sorry!

One possible rephrasing we came up with is "We need our liveness algorithm to consider the function's arguments live but the algorithm only considers instructions. To ensure that those arguments are considered as live, we make inputs to the function's first instruction."

Reply:
I had a go using my own words and came up with 395a946d607a06c5738115be354e447a5e5e13a9.

Maybe I spelled it out too hard?

Reply:
Works for me!